### PR TITLE
Add tests for CLEAR COLUMN IN PARTITION in SummingMergeTree

### PR DESCRIPTION
**Pull request by CarlosFelipeOR**
Summary                                                                                                                                                                                                                                               
                                                          
- Add engines/tests/summing_merge_tree/ test suite covering SummingMergeTree zero-row deletion behavior                                                                                                                                               
- Tests cover the regression found in ClickHouse upstream: https://github.com/ClickHouse/ClickHouse/issues/101953
- Two scenarios xfailed pending upstream fix:                                                                                                                                                                                                         
  - CLEAR COLUMN IN PARTITION does not trigger zero-row deletion after OPTIMIZE TABLE FINAL (regression since 25.8)                                                                                                                                   
  - CLEAR COLUMN validation inconsistency between explicit and auto-detected summing columns                                                                                                                                                          
                                                                                                                                                                                                                                                      
Test plan                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                      
- zero_row_deletion_with_update — passes on all versions                                                                                                                                                                                              
- zero_row_deletion_with_clear_column — xfail on >=25.8 (upstream issue)
- clear_column_validation_consistency — xfail on all versions (upstream issue)

### CODE REVIEW

File: engines/tests/summing_merge_tree/summing_merge_tree.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    if node is None:
+        node = self.context.cluster.node("clickhouse1")
+
+    name = f"summing_zero_update_{getuid()}"
+
+    try:
+        with Given("I create SummingMergeTree table with partition key"):
+            node.query(
+                f"CREATE TABLE {name} (v UInt64, p UInt64, c UInt64)"
```
Comment (Selfeer) [inline engines/tests/summing_merge_tree/summing_merge_tree.py]: It's best to reuse existing create table steps from `helpers/create` instead of creating a new query here - so the tests stay modular.

File: engines/tests/summing_merge_tree/summing_merge_tree.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                f" ENGINE = SummingMergeTree PARTITION BY p ORDER BY v"
+            )
+
+        with When("I insert data"):
+            node.query(f"INSERT INTO {name} VALUES (1, 1, 100)")
```
Comment (Selfeer) [inline engines/tests/summing_merge_tree/summing_merge_tree.py]: Same here - I would just crete a specific insert step either  here or in `steps.py` and call that step here

File: engines/tests/summing_merge_tree/summing_merge_tree.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            node.query(f"INSERT INTO {name} VALUES (2, 2, 200)")
+
+        with And("I update summing column to zero"):
+            node.query(f"ALTER TABLE {name} UPDATE c = 0 WHERE v = 1")
```
Comment (Selfeer) [inline engines/tests/summing_merge_tree/summing_merge_tree.py]: We also have alter steps under `helpers/alter`

Comment (Selfeer) [reply engines/tests/summing_merge_tree/summing_merge_tree.py]: https://github.com/Altinity/clickhouse-regression/blob/main/helpers/alter/update.py

File: engines/tests/summing_merge_tree/summing_merge_tree.py (hunk 1 above)
Comment (Selfeer) [reply engines/tests/summing_merge_tree/summing_merge_tree.py]: You probably can reuse this: https://github.com/Altinity/clickhouse-regression/blob/086674054b7a715ef384e4a6064fdd2a0f5c760b/helpers/create.py#L201

**Comment by Selfeer** [comment]
Please review that my changes didn't break your tests - afterwards it's approved from my side

**Comment by Selfeer** [comment]
The test also lacks requirements
